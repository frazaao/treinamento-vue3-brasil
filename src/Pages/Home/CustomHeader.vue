<template>
  <header class="h-[700px] bg-brand-main w-full flex justify-center">
    <div class="header-group">
      <div class="flex justify-between py-10 items-center">
        <picture class="w-28 lg:w-36">
          <img
            class="w-full"
            src="../../assets/images/logo_white.png"
            alt="Logotipo"
          />
        </picture>
        <nav>
          <ul class="flex">
            <li>
              <button
                class="hover:brightness-95 transition-all px-6 py-2 font-bold rounded-full text-white focus:outline-none"
                @click="() => emit('create-account')"
              >
                Crie uma conta
              </button>
            </li>

            <li>
              <button
                class="hover:brightness-95 transition-all px-6 py-2 font-bold bg-white rounded-full text-brand-main focus:outline-none"
                @click="() => emit('login')"
              >
                Entrar
              </button>
            </li>
          </ul>
        </nav>
      </div>

      <div class="flex flex-col mt-28">
        <h1 class="text-4xl font-black text-white">
          Tenha um feedback. <br />
          E faça seus clientes mais <br class="hidden lg:inline-block" />
          felizes!
        </h1>
        <p class="text-lg font-regular text-white">
          Receba ideias, reclamações e feedbacks com um
          <br class="hidden lg:inline-block" />
          simples widget na página.
        </p>
        <div>
          <button
            class="hover:scale-105 transition-all mt-10 px-6 py-2 font-bold bg-white rounded-full text-brand-main focus:outline-none"
          >
            Crie uma conta grátis
          </button>
        </div>
      </div>
    </div>
  </header>
</template>

<script>
export default {
  setup(_, { emit }) {
    return { emit };
  },
};
</script>

<style lang="postcss" scoped>
.header-group {
  @apply flex flex-col w-4/5 max-w-6xl;
}

@media (min-width: 600px) {
  .header-group {
    background-image: url("../../assets/images/blue_balloons.png");
    background-size: 622px;
    background-position: 90% 100%;
    background-repeat: no-repeat;
  }
}
</style>
