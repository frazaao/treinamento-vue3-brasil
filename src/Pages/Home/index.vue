<template>
  <CustomHeader @create-account="handleAccountCreate" @login="handleLogin" />
  <Contact />
  <footer class="flex justify-center py-10 bg-brand-gray">
    <p class="font-medium text-center text-gray-800">Feedbacker 2021</p>
  </footer>
</template>

<script>
import { onMounted } from "vue";
import { useRouter } from "vue-router";
import useModal from "../../hooks/useModal";
import CustomHeader from "./CustomHeader.vue";
import Contact from "./Contact.vue";

export default {
  name: "Home-index",
  components: {
    CustomHeader,
    Contact,
  },
  setup() {
    const router = useRouter();
    const modal = useModal();

    onMounted(() => {
      const token = window.localStorage.getItem("token");

      if (token) {
        router.push({ name: "Feedbacks" });
      }
    });

    function handleLogin() {
      modal.open({
        component: "ModalLogin",
      });
    }

    function handleAccountCreate() {
      modal.open({
        component: "ModalCreateAccount",
      });
    }

    return {
      handleLogin,
      handleAccountCreate,
    };
  },
};
</script>
