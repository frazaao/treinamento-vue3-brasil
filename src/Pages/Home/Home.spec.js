import Home from "./index.vue";
import { routes } from "../../router/index";

import { createRouter, createWebHistory } from "vue-router";
import { mount } from "@vue/test-utils";

const router = createRouter({
  history: createWebHistory("/"),
  routes,
});

describe("<Home />", () => {
  it("Should render home correctly", async () => {
    router.push("/");
    await router.isReady();
    const wrapper = mount(Home, {
      global: {
        plugins: [router],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
