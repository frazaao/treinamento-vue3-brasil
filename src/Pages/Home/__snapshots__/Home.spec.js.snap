// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Should render home correctly 1`] = `
<header class="h-[700px] bg-brand-main w-full flex justify-center">
  <div class="header-group">
    <div class="flex justify-between py-10 items-center">
      <picture class="w-28 lg:w-36"><img class="w-full" src="" alt="Logotipo"></picture>
      <nav>
        <ul class="flex">
          <li><button class="hover:brightness-95 transition-all px-6 py-2 font-bold rounded-full text-white focus:outline-none" id="header-create-account-button"> Crie uma conta </button></li>
          <li><button class="hover:brightness-95 transition-all px-6 py-2 font-bold bg-white rounded-full text-brand-main focus:outline-none" id="header-login-button"> Entrar </button></li>
        </ul>
      </nav>
    </div>
    <div class="flex flex-col mt-28">
      <h1 class="text-4xl font-black text-white"> Tenha um feedback. <br> E faça seus clientes mais <br class="hidden lg:inline-block"> felizes! </h1>
      <p class="text-lg font-regular text-white"> Receba ideias, reclamações e feedbacks com um <br class="hidden lg:inline-block"> simples widget na página. </p>
      <div><button class="hover:scale-105 transition-all mt-10 px-6 py-2 font-bold bg-white rounded-full text-brand-main focus:outline-none" id="cta-create-account-button"> Crie uma conta grátis </button></div>
    </div>
  </div>
</header>
<div class="flex justify-center w-full">
  <div class="flex flex-col items-center w-4/5 max-w-6xl my-16">
    <h1 class="text-4xl font-black text-center text-gray-800"> Alguma dúvida? </h1>
    <p class="text-lg text-center text-gray800 font-regular"> Quer saber melhor como functiona e quais são os preços? </p>
    <div class="mt-10"><a href="mailto:" class="px-6 py-2 mt-10 font-bold text-white rounded-full bg-brand-main focus:outline-none hover:brightness-95 transition-all"> Nos mande um email! </a></div>
  </div>
</div>
<footer class="flex justify-center py-10 bg-brand-gray">
  <p class="font-medium text-center text-gray-800">Feedbacker 2021</p>
</footer>
`;
